src/commands/registerCommands.ts

```diff
@@ -328,6 +329,60 @@ function registerAssociationCommands(context: ExtensionContext, languageClient:
 
 }
 
+interface ConfigurationItemEdit {
+  section: string;
+  value: any;
+  editType: ConfigurationItemEditType;
+}
+
+enum ConfigurationItemEditType {
+  Add = 0,
+  Delete = 1
+}
+
+/**
+ * Register commands used for registering/unregistering catalog for command menu and CodeLens
+ *
+ * @param context the extension context
+ * @param languageClient the language server client
+ */
+function registerUpdateConfigurationCommands(context: ExtensionContext) {
+  // For CodeLens
+  context.subscriptions.push(commands.registerCommand(ClientCommandConstants.UPDATE_CONFIGURATION, resolveConfigurationItemEdit));
+}
+
+function resolveConfigurationItemEdit<T>(configurationItemEdit: ConfigurationItemEdit) {
+  const document = (window.activeTextEditor && window.activeTextEditor.document && window.activeTextEditor.document.languageId === 'xml') ? window.activeTextEditor.document : undefined;
+  const currentWorkspaceUri = getWorkspaceUri(document).toString();
+  const catalogFilePath = getDirectoryPath(configurationItemEdit.value).includes(currentWorkspaceUri) ? getRelativePath(currentWorkspaceUri, configurationItemEdit.value) : configurationItemEdit.value;
```

Comment:
It is not generic because it is linked to catalog. I wonder if this resolution could be done on server side. Or if it is not possible, I think configurationItemEdit should store a kind data to say that it is a file. when kind = file we can resolve the path like you did.